### PR TITLE
feat: add ExtensionLogger for convenient VSCode extension logging

### PR DESCRIPTION
**Pull request by Klaus-Keller**
## Description

Add `ExtensionLogger` to package `@sap-ux/logger` for convenient logging in VSCode extensions.
Also updated dependencies and fixed minor linting issues.

### CODE REVIEW

File: packages/logger/src/extension-logger/index.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    debug(message: string, ...args: any): void {
+        this.logWithArgs(LogLevel.Debug, message, ...args);
+    }
+
+    /**
+     * Log a trace message.
+     *
+     * @param message - log message
+     */
```
Comment (zdravko-georgiev) [inline packages/logger/src/extension-logger/index.ts:77]: `@param` args is missing 😉

File: packages/logger/src/extension-logger/index.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    trace(message: string, ...args: any): void {
+        this.logWithArgs(LogLevel.Silly, message, ...args);
+    }
+
+    /**
+     * Show the output channel.
+     */
+    show(): void {
```
Comment (zdravko-georgiev) [inline packages/logger/src/extension-logger/index.ts:85]: Does this opens the Output tab in VSCode?

**Review by zdravko-georgiev — APPROVED** [review]
Code is well structured and easy to understand
Very good test coverage
Changeset is OK
Did NOT tested manually

Approved with one small comment

File: packages/logger/src/extension-logger/index.ts (hunk 2 above)
Comment (Klaus-Keller) [reply packages/logger/src/extension-logger/index.ts:85]: Yes

File: packages/logger/src/extension-logger/index.ts (hunk 1 above)
Comment (Klaus-Keller) [reply packages/logger/src/extension-logger/index.ts:78]: @zdravko-georgiev thanks, fixed.

**Review by zdravko-georgiev — APPROVED** [review]
Code is well structured and easy to understand
Very good test coverage
Changeset is OK
Did NOT tested manually

**Review by devinea — APPROVED** [review]
Code changes look good. 
Changeset ✅ 
Package updates look good. 
code coverage is great.

**Pull request merged**